File: Java/tracker/tracker/src/main/java/com/tacker/tracker/models/Accelerator.java
package com.tacker.tracker.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

@Entity
@Table(name = "accelerator")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Accelerator {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;  // Changed from Id to id

    private String value;

    @OneToMany(mappedBy = "accelerator", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<OppAccelerator> oppAccelerators;  // Changed from oppAcceleratorSet and class name

}

File: DotNet/ModelTracKer/Models/Accelerator.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ModelTracKer.Models
{
    [Table("acclerator")]
    public class Accelerator
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string? Name { get; set; }

        public ICollection<opp_Accelerator> OppAccelerators { get; set; }
    }
}
